src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -1878,6 +1903,10 @@ public override BoundNode VisitParameter(BoundParameter node)
             {
                 CheckAssigned(node.ParameterSymbol, node.Syntax);
             }
+            else if (CurrentSymbol is SynthesizedRecordConstructor)
+            {
+                NoteRead(node.ParameterSymbol);
```

Comment:
>NoteRead [](start = 16, length = 8)

I would prefer us to not call a method that isn't meant to be called. It is fine to have it to do a little bit of extra work when we call it anyway, it is another thing to call it in a new place when we only interested in that additional extra work and the method does plenty of other things. If the goal is to reuse code, it can be extracted into a helper that can be called from here and from ```NoteRead```